app/router.js

```diff
@@ -8,13 +8,25 @@ const Router = Ember.Router.extend({
 
 Router.map(function() {
   this.route('recipes', function() {
+
     this.route('recipe', {path: '/:recipe_id'});
+
   });
   this.route('features', function() {
     this.route('article'), {path: '/:slug'};
   });
   this.route('loading');
   this.route('notfound', {path: '*path'});
```

Comment:
Loading and Not found routes need to be at the bottom of the list, as they are catchall routes.